src/main/java/carpetaddonsnotfound/mixins/FlowerBlock_ReplaceFlowersMixin.java

```diff
@@ -15,22 +14,22 @@
 
 @Mixin(FlowerBlock.class)
 public abstract class FlowerBlock_ReplaceFlowersMixin extends Block {
-  FlowerBlock_ReplaceFlowersMixin() {
-    super(Settings.of(Material.PLANT));
+  public FlowerBlock_ReplaceFlowersMixin(Settings settings) {
+    super(settings);
   }
 
   @Override
   public boolean canReplace(BlockState state, ItemPlacementContext context) {
     if (replaceableFlowers == ReplaceableFlowersOptions.FALSE) {
       return super.canReplace(state, context);
     }
-    //From AbstrackBlock.canReplace but without Material.isReplacable check.
+    //From AbstrackBlock.canReplace but without BlockState.isReplacable check.
     return context.getStack().isEmpty() || !context.getStack().isOf(this.asItem());
   }
 
   @Override
   public void onStateReplaced(BlockState state, World world, BlockPos pos, BlockState newState, boolean moved) {
-    if (replaceableFlowers != ReplaceableFlowersOptions.FALSE && newState.getMaterial() != Material.AIR && !moved) {
+    if (replaceableFlowers != ReplaceableFlowersOptions.FALSE && !newState.isAir() && !moved) {
```

Comment:
What is isAir() actually doing? are you sure it will work the same?

Reply:
`isAir()` checks if the `BlockState` object corresponds to an air block. We cannot use `Material` anymore as it was removed.

I tested each rule and using it like this meant it behaved just the same as before.

Reply:
Does it check for cave air, structural void etc?

Reply:
Good question, I'll check now

Reply:
It gets it from the passed in `Settings` class to the constructor. So whatever that is set to.